src/session.ts

```diff
@@ -0,0 +1,262 @@
+import IAjaxSuccess = utils.IAjaxSuccess;
+import IAjaxError = utils.IAjaxError;
+
+/**
+ * The url for the session service.
+ */
+var SESSION_SERVICE_URL = 'api/sessions';
+
+
+/**
+ * Notebook Identification specification.
+ */
+export
+interface INotebookId {
+  path: string;
+};
+
+
+/**
+ * Session Identification specification.
+ */
+export
+interface ISessionId {
+  id: string;
+  notebook: INotebookId;
+  kernel: kernel.IKernelId;
+};
+
+
+/**
+ * Session initialization options.
+ */
+export
+interface ISessionOptions {
+  notebookPath: string;
+  kernelName: string;
+  baseUrl: string;
+  wsUrl: string;
+};
+
+/**
+ * Session object for accessing the session REST api. The session
+ * should be used to start kernels and then shut them down -- for
+ * all other operations, the kernel object should be used.
+ **/
+export 
+class Session {
+
+  static statusChanged = new Signal<Session, string>();
+
+  /**
+   * GET /api/sessions
+   *
+   * Get a list of the current sessions.
+   */
+  static list(baseUrl: string): Promise<ISessionId[]> {
+    var sessionUrl = utils.urlJoinEncode(baseUrl, SESSION_SERVICE_URL);
+    return utils.ajaxRequest(sessionUrl, {
+      method: "GET",
+      dataType: "json"
+    }).then((success: IAjaxSuccess): ISessionId[] => {
+      if (success.xhr.status !== 200) {
+        throw Error('Invalid Status: ' + success.xhr.status);
+      }
+      if (!Array.isArray(success.data)) {
+        throw Error('Invalid Session list');
+      }
+      for (var i = 0; i < success.data.length; i++) {
+        validateSessionId(success.data[i]);
+      }
+      return success.data;
+      
+    });
+  }
+
+  /**
+   * Construct a new session.
+   */
+  constructor(options: ISessionOptions) {
+    this._notebookPath = options.notebookPath;
+    this._baseUrl = options.baseUrl;
+    this._wsUrl = options.wsUrl;
+    this._kernel = new kernel.Kernel(this._baseUrl, this._wsUrl,
+                                     options.kernelName);
+  }
+
+  /**
+   * Get the session kernel object.
+  */
+  get kernel() : kernel.Kernel {
+    return this._kernel;
+  }
+
+  /**
+   * POST /api/sessions
+   *
+   * Start a new session. This function can only successfully executed once.
+   */
+  start(): Promise<void> {
+    var url = utils.urlJoinEncode(this._baseUrl, SESSION_SERVICE_URL);
+    return utils.ajaxRequest(url, {
+      method: "POST",
+      dataType: "json",
+      data: JSON.stringify(this._model),
+      contentType: 'application/json'
+    }).then((success: IAjaxSuccess) => {
+      if (success.xhr.status !== 201) {
+        throw Error('Invalid response');
+      }
+      validateSessionId(success.data);
+      this._kernel.start(success.data.kernel);
+      this._handleStatus('kernelCreated');
+    }, (error: IAjaxError) => {
+      this._handleStatus('kernelDead');
+    });
+  }
+
+  /**
+   * GET /api/sessions/[:session_id]
+   *
+   * Get information about a session.
+   */
+  getInfo(): Promise<ISessionId> {
```

Comment:
Do I read this correctly: the only info you can get from the session is it's id, but the id is provided as part of the ajax request?